lib/Feature/FeatureModelWriter.cpp

```diff
@@ -0,0 +1,314 @@
+#include "vara/Feature/FeatureModelWriter.h"
+#include "vara/Feature/FeatureModel.h"
+
+#include "XmlConstants.h"
+
+#include "libxml/xmlwriter.h"
+
+#define CHECK_RC                                                               \
+  if (RC < 0) {                                                                \
+    return RC;                                                                 \
+  }
+
+namespace vara::feature {
+
+static constexpr xmlChar INDENTATIONSTRING[] = "  ";
+static constexpr char ENCODING[] = "UTF-8";
+
+int FeatureModelXmlWriter::writeFeatureModel(std::string Path) {
+  int RC;
+  xmlTextWriterPtr Writer;
+
+  Writer = xmlNewTextWriterFilename(Path.data(), 0);
+  if (Writer == nullptr) {
+    return false;
+  }
+
+  RC = writeFeatureModel(Writer);
+  xmlFreeTextWriter(Writer);
+  return RC;
+}
+
+std::optional<std::string> FeatureModelXmlWriter::writeFeatureModel() {
+  int RC;
+  xmlTextWriterPtr Writer;
+  xmlDocPtr Doc;
```

Comment:
Here you should use unique_ptr with a custom deleter, examples can be found in the parser code